script/testing/util/periodic_task.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python3
+from threading import Timer
+
+
+class PeriodicTask:
+    """
+    A utility class for simple periodic tasks with Timer threads
+    """
+    def __init__(self, interval, function, *args, **kwargs):
+        """
+        Initialize the PeriodicTask object and schedule the initial Timer
+        """
+        self._timer = None
+        self.function = function
+        self.interval = interval
+        self.args = args
+        self.kwargs = kwargs
+        self.is_running = False
+        self.start()
+
+    def _run(self):
+        """
+        Execute the periodic task and rescheule the next Timer thread
+        """
+        self.is_running = False
```

Comment:
This isn't intuitive. By setting is_running = False the if statement in `start` is kinda useless

Reply:
No, the `is_running` means whether the PeriodicTask has any existing Timer thread already schedule or not. It is important not to double schedule a Timer which may mess up the interval and lose track of the lifecycle of one thread (but many more threads whenever the `.start()` is called). It is a safeguard measurement.